### PR TITLE
[DRAFT] Added retry library

### PR DESCRIPTION
**Pull request by PakDLiu**
* Added retry library
* Refactored out logic for `_actionExecuteSql` into its own method `actionExecuteSql`
* Added retry to `actionExecuteSql`
* `actionExecuteSql` will throw execption rather than return booleans

### CODE REVIEW

**Comment by PakDLiu** [comment]
logs:
```
DEBUG    [SapHana/hanaPrimary.HostConfig] verifying if check is enabled
DEBUG    [SapHana/hanaPrimary.HostConfig] verifying if check is due to be run
DEBUG    [SapHana/hanaPrimary.HostConfig] lastRunLocal=None; frequencySecs=60; currentLocal=2020-05-01 21:11:11.851752
INFO     [SapHana/hanaPrimary.HostConfig] executing all actions of check
DEBUG    [SapHana/hanaPrimary.HostConfig] actions=[{'parameters': {'sql': 'SELECT * FROM SYS.M_LANDSCAPE_HOST_CONFIGURATION ORDER BY HOST_ACTIVE DESC, INDEXSERVER_ACTUAL_ROLE ASC'}, 'type': 'ExecuteSql'}, {'type': 'ParseHostConfig'}]
DEBUG    [SapHana/hanaPrimary.HostConfig] calling action _actionExecuteSql
INFO     [SapHana/hanaPrimary.HostConfig] connecting to HANA and executing SQL
INFO     [SapHana/hanaPrimary.HostConfig] establishing connection with HANA instance
DEBUG    [SapHana/hanaPrimary.HostConfig] no host config has been persisted yet, using user-provided host
DEBUG    hostsToTry=['10.1.2.4']
WARNING  [SapHana/hanaPrimary.HostConfig] could not connect to HANA node 10.1.2.4:30113 ((-10709, "Connection failed (RTE:[89006] System call 'connect' failed, rc=111:Connection refused {10.1.2.4:30113} (10.1.2.4:30113))"))
ERROR    [SapHana/hanaPrimary.HostConfig] unable to connect to any HANA node (hosts to try=['10.1.2.4'])
WARNING  Failed _getHanaConnection, retrying in 1 seconds...
INFO     [SapHana/hanaPrimary.HostConfig] connecting to HANA and executing SQL
INFO     [SapHana/hanaPrimary.HostConfig] establishing connection with HANA instance
DEBUG    [SapHana/hanaPrimary.HostConfig] no host config has been persisted yet, using user-provided host
DEBUG    hostsToTry=['10.1.2.4']
WARNING  [SapHana/hanaPrimary.HostConfig] could not connect to HANA node 10.1.2.4:30113 ((-10709, "Connection failed (RTE:[89006] System call 'connect' failed, rc=111:Connection refused {10.1.2.4:30113} (10.1.2.4:30113))"))
ERROR    [SapHana/hanaPrimary.HostConfig] unable to connect to any HANA node (hosts to try=['10.1.2.4'])
WARNING  Failed _getHanaConnection, retrying in 2 seconds...
INFO     [SapHana/hanaPrimary.HostConfig] connecting to HANA and executing SQL
INFO     [SapHana/hanaPrimary.HostConfig] establishing connection with HANA instance
DEBUG    [SapHana/hanaPrimary.HostConfig] no host config has been persisted yet, using user-provided host
DEBUG    hostsToTry=['10.1.2.4']
WARNING  [SapHana/hanaPrimary.HostConfig] could not connect to HANA node 10.1.2.4:30113 ((-10709, "Connection failed (RTE:[89006] System call 'connect' failed, rc=111:Connection refused {10.1.2.4:30113} (10.1.2.4:30113))"))
ERROR    [SapHana/hanaPrimary.HostConfig] unable to connect to any HANA node (hosts to try=['10.1.2.4'])
INFO     [SapHana/hanaPrimary.HostConfig] error executing action _actionExecuteSql, skipping remaining actions
DEBUG    [SapHana/hanaPrimary.HostConfig] calling action _actionParseHostConfig
INFO     [SapHana/hanaPrimary.HostConfig] parsing HANA host configuration and storing it in provider state

```

File: sapmon/setup/configureMonitorVM.sh [hunk 1]
```diff
@@ -44,3 +44,5 @@ ExecuteCommand "pip3 install azure-identity"
 ExecuteCommand "pip3 install azure-keyvault-secrets"
 # Install prometheus_client
 ExecuteCommand "pip3 install prometheus_client"
+# Install retry
+ExecuteCommand "pip3 install retry"
```
Comment (persiaAziz) [inline sapmon/setup/configureMonitorVM.sh:48]: Please add it to the docker image for v2.0 as well

File: sapmon/payload/provider/saphana.py [hunk 2]
```diff
@@ -361,10 +362,25 @@ def _actionExecuteSql(self,
...
+   def _actionExecuteSql(self,
+                         sql: str,
+                         isTimeSeries: bool = False,
+                         initialTimespanSecs: int = 60) -> bool:
+      try:
+         retry_call(
+            self.actionExecuteSql,
+            fargs=[sql, isTimeSeries, initialTimespanSecs],
+            tries=3,
+            delay=1,
```
Comment (persiaAziz) [inline sapmon/payload/provider/saphana.py:378]: is delay the initial delay here? and then backoff takes place for further retry?

Comment (persiaAziz) [reply sapmon/payload/provider/saphana.py:378]: Don't we want to take provider type into consideration as well?

**Pull request closed without merging**